pkg/morph/client/netmap/netmap.go

```diff
@@ -107,6 +145,103 @@ func (c *Client) EpochSnapshot(args EpochSnapshotArgs) (*EpochSnapshotValues, er
 	}, nil
 }
 
+func (c *Client) Candidates(_ GetNetMapCandidatesArgs) (*GetNetMapCandidatesValues, error) {
+	prms, err := c.client.TestInvoke(
+		c.netMapCandidatesMethod,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("could not perform test invocation (%s): %w", c.netMapCandidatesMethod, err)
+	}
+
+	candVals, err := peersWithStateFromStackItems(prms, c.netMapCandidatesMethod)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse contract response: %w", err)
+	}
+
+	return candVals, nil
+}
+
+func peersWithStateFromStackItems(stack []stackitem.Item, method string) (*GetNetMapCandidatesValues, error) {
+	if ln := len(stack); ln != 1 {
+		return nil, fmt.Errorf("unexpected stack intem count (%s): %d", method, ln)
+	}
+
+	netmapNodes, err := client.ArrayFromStackItem(stack[0])
+	if err != nil {
+		return nil, fmt.Errorf("could not get stack item array from stack item (%s): %w", method, err)
+	}
+
+	res := &GetNetMapCandidatesValues{
+		NetmapNodes: make([]*PeerWithState, 0, len(netmapNodes)),
+	}
+
+	for i := range netmapNodes {
+		node, err := peerWithStateFromStackItem(netmapNodes[i])
+		if err != nil {
+			return nil, fmt.Errorf("could not parse stack item (Peer #%d): %w", i, err)
+		}
+
+		res.NetmapNodes = append(res.NetmapNodes, node)
+	}
+
+	return res, nil
+}
+
+func peerWithStateFromStackItem(prm stackitem.Item) (*PeerWithState, error) {
+	if prm.Type() != stackitem.StructT {
```

Comment:
Why do we need explicit `stackitem.StructT` which is supported by `client.ArrayFromStackItem`?

Reply:
an additional layer of checks since there are too deep nesting of structures(there was no nested structs parsing before that PR)